src/work-allocation-2/containers/work-case-list/work-case-list.component.spec.ts

```diff
@@ -0,0 +1,459 @@
+        { provide: FeatureToggleService, useValue: mockFeatureToggleService }
+      ]
+    }).compileComponents();
+    fixture = TestBed.createComponent(WrapperComponent);
+    wrapper = fixture.componentInstance;
+    component = wrapper.appComponentRef;
+
+    wrapper.cases = getCases();
+    wrapper.fields = getFields();
+    wrapper.caseServiceConfig = getCaseService();
+    wrapper.casesTotal = 2;
+    wrapper.pagination = {
+      page_number: 1,
+      page_size: 10
+    };
+    mockWorkAllocationService.getCase.and.returnValue(of({}));
+    mockFeatureToggleService.isEnabled.and.returnValue(of(true));
+    fixture.detectChanges();
+  }));
+
+  it('should return the fields as an array with a \'manage\' entry, so that we can' +
+    'display the manage column in the table.', async () => {
+
+    const fields = ['caseReference', 'caseName', 'caseCategory', 'location', 'case', 'dueDate'];
+    const fieldsWithManage = [...fields, 'manage'];
+
+    expect(component.addManageColumn(fields)).toEqual(fieldsWithManage);
+  });
+
+  it('should return the columns to be displayed by the Angular Component Dev Kit table.', async () => {
+
+    // create mock getDisplayedColumn variables
+    const caseFieldConfig = getFields();
+    const fields = caseFieldConfig.map(field => field.name);
+    const displayedColumns = component.addManageColumn(fields);
+
+    // test actual function against mock variables
+    expect(component.getDisplayedColumn(caseFieldConfig)).toEqual(displayedColumns);
+
+  });
+
+  it('should take in the field name and trigger a new Request to the API to get a sorted result set.', async () => {
+
+    // mock the emitter and dispatch the connected event
+    spyOn(component.sortEvent, 'emit');
+    const element = fixture.debugElement.nativeElement;
+    const button = element.querySelector('#sort_by_caseName');
+    button.dispatchEvent(new Event('click'));
+    fixture.detectChanges();
+
+    // check the emitter had been called and that it gets called with the first field which is caseReference (caseId)
+    expect(component.sortEvent.emit).toHaveBeenCalled();
+    expect(component.sortEvent.emit).toHaveBeenCalledWith('caseName');
+  });
+
+  it('reset sort button is hidden by default', async () => {
+    expect(component.showResetSortButton).toBeFalsy();
+  });
+
+  it('show reset sort button after clicking column header', async () => {
+
+    /// mock the emitter and dispatch the connected event
+    spyOn(component.sortEvent, 'emit');
+    const element = fixture.debugElement.nativeElement;
+    const button = element.querySelector('#sort_by_caseName');
+    button.dispatchEvent(new Event('click'));
+    component.sortedBy = { fieldName: 'caseName', order: SortOrder.DESC };
+    fixture.detectChanges();
+
+    // check the emitter had been called and that it gets called with the new field defined which is caseName
+    expect(component.sortEvent.emit).toHaveBeenCalled();
```

Comment:
_Nitpick:_
These two are redundant checks as they're tested for in the previous unit test.

Reply:
You're right. Removed.